sync/src/chain.rs

```diff
@@ -209,6 +212,8 @@ pub struct ChainSync {
 	max_download_ahead_blocks: usize,
 	/// Network ID
 	network_id: U256,
+	/// Transactions Queue
+	transaction_queue: Mutex<TransactionQueue>,
```

Comment:
I was asking about thread safety, not multithreaded processing. Already checked that it is not currently thread safe, so mutex is fine here. As for parallelization, I don't currently see the need for that with the transaction volume we have on the main network. Can be added later.